UDX-7921: Initial helper files and scripts for python users

adam-poppenhagen opened the pull request:
Signed-off-by: Adam Poppenhagen <adam.j.poppenhagen@seagate.com>

adam-poppenhagen committed:
UDX-7921: Initial helper files and scripts for python users

Signed-off-by: Adam Poppenhagen <adam.j.poppenhagen@seagate.com>

ux-joy commented:
Hi Adam - you don't usually need my review for code-only changes. Unless it affects the UI, which I don't think this does. Thank you for including me anyway! :-)

sherohmanA commented:
@adam-poppenhagen, do you still need to push an update to remove your hostname from these changes? Was waiting for an update there, but if no further update I'll go ahead and approve

adam-poppenhagen commented:
Sorry everyone. I'm going to polish everything and get it ready to merge. I'll reach out again when I have everything good to go

adam-poppenhagen commented:
![image](https://user-images.githubusercontent.com/91157605/176218274-c549189f-5a1d-4fec-b87f-bdd8b3208db9.png)

Intended output for Python script

adam-poppenhagen committed:
Deleted personal info and added Documentation

adam-poppenhagen committed:
deleted one more email I left in and changed new guy to new user

adam-poppenhagen commented:
Ok, I think I've done this properly and am fully ready to merge branches. I have deleted all passwords and usernames and added documentation. In theory a customer could literally replace everything at the top of the files in brackets with their information and run the script without problems as I have done in demos before. This can also provide a template for future API scripts so if there's any input as to formatting we can resolve that too.

jelyware commented on apiproject.py:
```diff
@@ -0,0 +1,106 @@
+pilotUtils.printBold('\n----- USERS -----')
+responseData = pilotUtils.getUsers(HOST, token)
+for i in responseData['data']:
+  print(i['name'])
```
perhaps more informative print statements, example:
`print('User name: %s\n' % i['name'])`

adam-poppenhagen replied:
I like this. I think it's a much more streamlined output

adam-poppenhagen replied:
I also had the thought of having the function do that behind the scenes but I thought it might be beneficial for the user to see that the output was parsed

jelyware commented on nodejs/Users.js:
```diff
@@ -0,0 +1,43 @@
+  try {
+    let jwt = await pilotUtils.login();
+    console.log('Login Token: ' + jwt);
+    // console.log(base_url + USERS_URL);
```
remove comment?

adam-poppenhagen replied:
I removed the comment and cleaned up User.js. It isn't super functional but it is the beginning of a separate file that imports pilotUtils.js should an end user choose to do it that way. It am not sure whether we should just delete the whole file or put the time in to fix it up and make it fully functional

jelyware commented on nodejs/Users.js:
```diff
@@ -0,0 +1,43 @@
+const { exit } = require('process')
+
+const args = process.argv.slice(2);
+const uuid = '156b9cujydh6dq9s44dubnsg7r'; //args[0];
```
do we want these hardcoded in here? or passed in as args?  this looks like it was just in place for testing :-)

adam-poppenhagen replied:
I see. I forgot that I had played with that file as well. Good catch!

jelyware commented on nodejs/getDevices.js:
```diff
@@ -1,123 +1,123 @@
+const { exit } = require('process');
+
+const args = process.argv.slice(2);
+const uuid = '156b9cujydh6dq9s44dubnsg7r'; //args[0];
```
you want to put the args[0], args[1], etc. back in here too :-)

jelyware commented on nodejs/pilotUtils.js:
```diff
@@ -0,0 +1,351 @@
+// const { exit } = require('process');
+
+// const args = process.argv.slice(2);
+const uuid = '<Your UUID>'; //args[0];
```
again these can be set by command parameters args[0], args[1], etc.

adam-poppenhagen replied:
For this I wasn't sure what to do. I wanted to leave those to show that this could be called from the command line but also you could include the variables in the file and have like a single button push kind of functionality. Would it be more common for something like this to run via cli?

adam-poppenhagen replied:
However, for getDevices.js specifically I have reverted it to cli args

jelyware replied:
If the user experience is actually to alter the script itself withOUT cli args then you should keep it how you have it.  You'll need to make sure this is communicated also in the user documentation :-)  Btw, is Ben W. doing that documentation for you?

adam-poppenhagen replied:
I've been thinking about your comment here and I have a meeting with Corey, Morgan, Dylan and Dave tomorrow and I want to address this because you're absolutely right that these need to be standardized so I want to bring this up to them and see what they want out of it

jelyware replied:
ok I will be prepared to re-review if changes come out of your upcoming meeting.  Thanks Adam!

jelyware reviewed (commented):
Hey Adam, great job!

adam-poppenhagen committed:
Changes suggested by Jessica

adam-poppenhagen committed:
removed several testing files

adam-poppenhagen commented:
Awesome comments by Jessica. I really appreciate it. I forgot I had some testing files that shouldn't be public so I got rid of some and I'll continue to dig to make sure I didn't leave any others in

kbertelson commented on python/src/pilotUtils.py:
```diff
@@ -1,87 +1,288 @@
+def print_json(output):
+  pp.pprint(output)
+
+def print_bold(output):
+  print('\033[1m' + output + '\033[0m')
```
I kinda like the colorama approach which allows you to sprinkle color easily whenever you're formatting a string.

```
from colorama import Fore
from colorama import Style

def echo_error(text: str):
    """Print error msg to stderr"""
    print(f"{Fore.RED}ERROR:{Style.RESET_ALL} {text}", file=sys.stderr)
```

Your code represents example files to end-users, so we may not want to include a dependency on colorama. So, I could see some reluctance to include it.

Will most end-users expect indentation to be 4 spaces rather than our 2? Did the Pylint extension for vscode complain about 2 space indentation?

adam-poppenhagen replied:
I have never used colorama before so that is something I will definitely play with. print_bold was something that existed in the module before I got to it so I just went with it but I do think it's very important to make sure that the output looks logical and thought out and the ability to color outputs could be really useful as some of the prints can get messy. We do already have a dependency on Axios for Nodejs so one more might not be a big deal.

I think you might be on to something with the indentations. I have always used 4, but the 2 space indent again was something preestablished in the module pilotUtils before I started expanding on it and so I just went with it because I thought maybe it was a standard I was unaware of or something. Vscode also never raised any issues with it so I left it as is. But I do agree that end users may expect 4. We have a meeting tomorrow and I will bring this up and see if the 2 space was intentional

jelyware replied:
agree with Ken and you about the 4-space vs. 2-space indentation for python scripts and that users will expect the 4-spaces...  I'll be standing by to re-review if needed thanks!

jelyware reviewed (approved):
thanks for making those changes Adam!
